### ISSUE TITLE
Project Discussion

### ISSUE THREAD

**Issue by corytortoise** [TurtleTeam/MurderMysteryPE#1]
## Project Discussion
Let's use this issue to discuss anything project-related, including quick questions, contribution discussion, etc. For other issues, please make a separate issue in the tracker to prevent confusion.

**Comment by corytortoise** [comment]
Considering the amount of people that want to contribute, I believe we need to make a consensus on contribution guidelines. I suggest we use the same code formatting as [PMMP](https://github.com/pmmp/PocketMine-MP/blob/master/CONTRIBUTING.md#code-syntax), but I am open to any suggestions.

**Comment by jasonw4331** [comment]
This plugin is going to be a lot bigger a deal than I had anticipated

**Comment by ghost** [comment]
Can i put my own version here? Just to test it out and improves it?

**Comment by jasonw4331** [comment]
@imjayl you already have a version that works?

**Comment by ghost** [comment]
@jasonwynn10  Yes its easy to customize but the only problem is i didnt add the gold spawn thing

**Comment by jasonw4331** [comment]
link?

**Comment by ghost** [comment]
View your message at pmmp forum :)

**Comment by xBeastMode** [comment]
@jasonwynn10 can you create a proper issue with the total list of features?

**Comment by Kris-Driv** [comment]
Put @imjayl code into separate branch.

**Comment by jasonw4331** [comment]
He asks that the code is not to be made public, so we won't be able to copy it explicitly, but I think we may be able to use it as an example.

**Comment by Kris-Driv** [comment]
Also, make sure that we don't make too much code changes per commit. In project like this and at this stage, it can cause file conflicts a lot.

**Comment by corytortoise** [comment]
@Chris-Prime I agree.

**Comment by xBeastMode** [comment]
And if it's possible please use Git, it will automatically sync the repository for you, then you can do your part and commit it.

**Comment by jasonw4331** [comment]
I won't be able to really code very much until I can get my computer repaired, but I will be using PhpStorm once I do, so you all shouldn't have anything to worry about.

**Comment by Kris-Driv** [comment]
Main class name seems too plain, what about Launcher or MurderMystery?

**Comment by xBeastMode** [comment]
MurderMystery is good.

**Comment by Kris-Driv** [comment]
At some point we'll need to implement language files. What's your opinion on using namespace function which would look like this

```php
namespace TurtleTeam\MurderMystery;

function lang(string $key, ...$params): string {
   return Foo::getInstance()->getMessage($key, $params);
}

class Foo {
   
   public function onEnable() {
      $this->getLogger()->info(lang("plugin-enabling"));
   }
}
```

**Comment by xBeastMode** [comment]
I guess it's good but I think we should make it `static final` just for security.

**Comment by Sandertv** [comment]
People, is this plugin still going to be continued? I can do all the bodies part and stuff if you want.

**Comment by jasonw4331** [comment]
I have my own project running separately which has made more progress... I'm pretty sure this is discontinued, but if the others feel like working on it, they may feel free to do so

**Comment by corytortoise** [comment]
If somebody wants to take over ownership, let me know. I'm not worthy or or knowledgable enough to oversee this.
